src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java

```diff
@@ -563,53 +602,35 @@ private String transformOMAScore( String omaScore) {
     private String processMAFHeader(BufferedReader buffer) throws IOException, DaoException {
         GeneticProfile geneticProfile = DaoGeneticProfile.getGeneticProfileById(geneticProfileId);
         String line = buffer.readLine().trim();
+        Set<Integer> internalSampleIds = new HashSet<>();
         while (line.startsWith("#")) {
             Matcher seqSamplesMatcher = SEQUENCE_SAMPLES_REGEX.matcher(line);
             // line is of format #sequenced_samples: STABLE_ID STABLE_ID STABLE_ID STABLE_ID
             if (seqSamplesMatcher.find()) {
-                addSampleProfileRecords(getSequencedSamples(seqSamplesMatcher.group(1), geneticProfile));
+                internalSampleIds.addAll(getSequencedInternalSampleId(seqSamplesMatcher.group(1), geneticProfile));
             }
             line = buffer.readLine().trim();
         }
+        DaoSampleProfile.upsertSampleProfiles(internalSampleIds, geneticProfileId, genePanelId);
```

Comment:
I think upsertSampleProfileRecords is a better name. Or upsertSampleToProfileRecords , or upsertSampleVsProfileRecords . "Sampleprofiles" is confusing as it sounds like it is referring to the complete sample profile (i.e. the mutation data itself).

Reply:
upsertSampleToProfileMapping is perhaps the best name...

Reply:
I like `upsertSampleToProfileMapping` Renamed here https://github.com/cBioPortal/cbioportal-core/pull/48/commits/f120f5d14e4c2928309d89a041c8fdfb1920476a